### ISSUE TITLE
Empty `mail` user password

### ISSUE THREAD

**Issue by SelfAdjointOperator** [LukeSmithxyz/emailwiz#244]
Hey Luke,
I solved my old issue #239 and it's *not* what you might expect.
TLDR: the password for the `mail` user on my server in `/etc/shadow` was empty and valid!

I was still getting what I thought to be 'backscatter' so I checked `journalctl`. I was a little more observant and saw the line
`pam_unix(dovecot:auth): user [mail] has blank password; authenticated without it`.
I did some Googling and found a [related issue](https://github.com/linux-pam/linux-pam/issues/284), something to do with `/etc/shadow`.
I checked mine and low and behold
```
...
lp:*:18978:0:99999:7:::
mail::19194:0:99999:7:::
news:*:18978:0:99999:7:::
...
```
Having a look at `man 5 shadow` tells us that we must have a `*` or `!` after the first `:` to indicate that a user has no password, and is unavailable. However, the `mail` user didn't for whatever reason. Anyone was able to connect to my server via postfix / smtp and log in to the `mail` user with a blank password over [sasl](https://www.postfix.org/SASL_README.html) (I don't know much about sasl but postfix forwards it to dovecot via a unix socket and then uses pam to authenticate).

I've no idea what on earth made the `mail` user have a valid empty password, whether the server ISO came like that, or some later-installed package made it so, but I've certainly checked `/etc/shadow` for any other valid-empty-password users, and updated `mail` via `passwd`. This feels quite embarrassing, but it's worth mentioning. Thank goodness for key-only ssh or I'd be toast! (I may reinstall anyways idk).

So yeah I *think* I should be all good now, looks like it was nothing to do with postfix / dovecot / emailwiz, this glows lol.
Perhaps other people might want to check this on their own servers? Though no one else seems to have raised this issue before.
